src/js/components/NewsFeedPost.jsx

```diff
@@ -1,59 +1,67 @@
-import React from "react";
-import NewsFeedStore from "stores/NewsFeedStore";
+import React from 'react';
+import NewsFeedStore from 'stores/NewsFeedStore';
 import FaHeart from 'react-icons/lib/fa/heart';
 import FaRetweet from 'react-icons/lib/fa/retweet';
 
 export default class NewsFeedPost extends React.Component {
 
   constructor(props) {
     super(props);
-    // this.togglePictureDisplay = this.togglePictureDisplay.bind(this);
-    this.changeLike = this.changeLike.bind(this);
     this.state = this.props;
+    this.changeLike = this.changeLike.bind(this);
+    this.retweet = this.retweet.bind(this);
+    this.showImage = this.showImage.bind(this);
   }
 
-  // componentWillMount() {
-  //   NewsFeedStore.on("updated", this.getNewsFeedPosts);
-  //   NewsFeedStore.on("error", this.showError);
-  // }
+  // componentWillMount() {}
 
-  // componentWillUnmount() {
-  //   NewsFeedStore.removeListener("updated", this.getNewsFeedPosts);
-  //   NewsFeedStore.removeListener("error", this.showError);
-  // }
-
-  showError() {
-    console.log(NewsFeedStoreStore.error);
-  }
+  // componentWillUnmount() {}
 
   changeLike() {
     if (this.state.current_user_likes) {
-      NewsFeedStore.decrementLike();
+      NewsFeedStore.unlikePost(this.state.id);
+      // TODO: Potentially move this logic to be based on emitted action?
       this.setState({
         current_user_likes: false,
+        like_count: (parseInt(this.state.like_count) - 1).toString(),
```

Comment:
This is the process of the user unliking the post. So I remove one from the count on the post if they had previously liked it. I was wondering if I should do this optimistically as above so the user sees feedback or actually dispatch it when I get a response and make the change then